azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -291,11 +291,32 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 							Optional: true,
 						},
 
+						"ip_forwarding": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
 						"network_security_group_id": {
 							Type:     schema.TypeString,
 							Optional: true,
 						},
 
+						"dns_settings": {
+							Type:     schema.TypeSet,
```

Comment:
Your tests are using the `.0` list syntax while this is a `TypeSet`. I suggest changing this to a a `TypeList`

Reply:
using the same approach, I would need to convert the `network_profile` to a `TypeList` to allow the test to work. Is this acceptable? I feel that we are using the wrong data type for to be compatible with the test harness. Both `dns_settings` and `network_profile` should be Set with MaxItem = 1 in my opinion.

Reply:
Yep! I would prefer a `TypeList` here for the above reasons. A type set is only really useful when you have >1 item, order doesn't matter and duplicates are not allowed

Reply:
Fixed, `network_profile` was kept a `TypeSet` while `dns_settings` was converted to a TypeList`, the ACCtests were updated accordingly